numpy/random/tests/test_extending.py

```diff
@@ -24,28 +26,32 @@
     import cython
 except ImportError:
     cython = None
+else:
+    cython_ver = cython.__version__.split('.')
+    # Cython 0.29.14 is required for Python 3.8 and there are
+    # other fixes in the 0.29 series that are needed even for earlier
+    # Python versions.
+    # Note: keep in sync with the one in pyproject.toml
+    if len(cython_ver) < 3 or cython_ver < ['0', '29', '14']:
+        # too old or wrong cython, skip the test
+        cython = None
 
 @pytest.mark.skipif(cython is None, reason="requires cython")
-def test_cython():
-    curdir = os.getcwd()
-    argv = sys.argv
-    examples = (os.path.dirname(__file__), '..', '_examples')
-    try:
-        os.chdir(os.path.join(*examples))
-        sys.argv = argv[:1] + ['build']
-        with warnings.catch_warnings(record=True) as w:
-            # setuptools issue gh-1885
-            warnings.filterwarnings('always', '', DeprecationWarning)
-            from numpy.random._examples.cython import setup
-    finally:
-        sys.argv = argv
-        os.chdir(curdir)
+@pytest.mark.slow
+@pytest.mark.skipif(sys.platform == 'win32', reason="cmd too long on CI")
+def test_cython(tmp_path):
+    examples = os.path.join(os.path.dirname(__file__), '..', '_examples')
+    base = os.path.dirname(examples)
+    shutil.copytree(examples, tmp_path / '_examples')
+    subprocess.check_call([sys.executable, 'setup.py', 'build'],
+                          env=os.environ,
```

Comment:
The docs say that when `env` is passed,

> these are used instead of the default behavior of inheriting the current process’ environment

Unless you want any env vars inserted by pytest itself (are there any?), you can just remove this line.
```suggestion
```

Reply:
ok